README.md

```diff
@@ -1,2 +1,105 @@
 # elixir-base58-encode
 https://hex.pm/packages/b58
+
+
+This module provides the `Base58Encode.encode/1` function which takes an **Elixir binary** and returns its base58 String representation.
+
+See the section [What is an Elixir binary?](#What-is-an-elixir-binary) for more information about the binary type in Elixir.
+
+## How to use the module
+
+1. Add the package to you dependencies
+
+  ```
+  defp deps do
+    [b58: "~> 0.1.0"]
+  end
+  ```
+
+  and run `mix deps.get`
+
+2. Call the `encode` function with a binary as parameter:
+
+ ```
+ Base58Encode.encode("foo")
+ "bQbp"
+ ```
+
+ if the parameter is not an Elixir binary the function will return `:error`. In
+ this example 42 is an Integer and not a binary (`is_binary(42)` will return `false`)
+
+ ```
+ Base58Encode.encode(42)
+ :error
+ ```
+
+ See `example/Example.exs` file for a simple example on how to use the module.
+ You can also run this example with `mix run example/Example.exs`
+
+## What is base58?
+
+`base58` provides a way to represent an **integer to a string** where the characters of the string are selected from the following list:  
+
+`123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz`
+
+The idea is to map each character of the set to a value: 0 to 1, 2 to 2 ... 57 to z.
+
+The integer is then converted to the base58 list of codes. For example 65
+is represented as a list of two codes 1 and 7:
+```
+65 % 58 = 7 (65 modulo 58 is 7 (65 - 58 * 1 = 7))
+1 % 58 = 1
+
+So 65 is represented by 1 7 codes
+```
```

Comment:
This section isn't too clear to me. Why do we use modulo? Where does `1 % 58` come from?

Reply:
I didn't want had too much details on how to convert number from one base to another but I can try to describe the steps a bit more :+1:

Reply:
Oh I liked this section actually.

Maybe add an example with `rem/2` function (I know it does the same thing as modulo but maybe it will help to make it more clear to some people)? Just a thought